### PR TITLE
Added option for stdout

### PR DESCRIPTION
**Pull request by RichardLitt**
By adding the `-s` or `--stdout` option, you can print to stdout now. Not the most beautiful PR, but I think it is pretty efficient and it works for me. Let me know if I can edit anything!

Related to issue #100

### CODE REVIEW

**Comment by thlorenz** [comment]
Could you please add a test for this?
You could use `child_process` to launch the doctoc cli tool and check for the output, i.e. make sure it prints if `-s,--stdout` is specified but writes to the file otherwise.
Aside from that LGTM

**Comment by RichardLitt** [comment]
@thlorenz That should do it! 

And thank you! I love doctoc, btw; have been using it for a long time now on many different repos. Amazing thing.

File: test/transform-stdout.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+test('\nshould print to stdout with --stdout option', function (t) {
+
+    exec('node doctoc.js test/fixtures/readme-with-custom-title.md --stdout', (error, stdout, stderr) => {
+      if (error) {
+        console.error(`exec error: ${error}`);
+        return;
+      }
+      t.deepEqual(stdout,
+        `
+DocToccing single file "test/fixtures/readme-with-custom-title.md" for github.com.
```
Comment (thlorenz) [inline test/transform-stdout.js]: Huge fan of template literals, but those won't run with older node versions and I'd like to keep these running.
If you could please pull these out into files in the [fixtures](https://github.com/thlorenz/doctoc/tree/master/test/fixtures) and read them in instead, we'd highly appreciate it :).
I know it's more work, but then again nothing else in this repo uses non ES5 features, so for a bit I'd like to stay backwards compat.

**Comment by thlorenz** [comment]
@RichardLitt almost there .. thanks for your patience :)

**Comment by RichardLitt** [comment]
Saw that coming, should have done that myself. :) Fixed now.

File: test/transform-stdout.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+test('\nshould print to stdout with --stdout option', function (t) {
+
+    exec('node doctoc.js test/fixtures/readme-with-custom-title.md --stdout', function (error, stdout, stderr) {
+      if (error) {
+        console.error('exec error: ', error);
+        return;
+      }
+      t.deepEqual(stdout
+        , require('fs').readFileSync(__dirname + '/fixtures/stdout.md', 'utf8')
+        , 'spits out the correct table of contents')
```
Comment (thlorenz) [inline test/transform-stdout.js]: final nit: please require `fs` at the top of file.

File: test/transform-stdout.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+test('\nshould print to stdout with -s option', function (t) {
+
+    exec('node doctoc.js test/fixtures/readme-with-custom-title.md -s', function (error, stdout, stderr) {
+      if (error) {
+        console.error('exec error: ', error);
+        return;
+      }
+      t.deepEqual(stdout
+        , require('fs').readFileSync(__dirname + '/fixtures/stdout.md', 'utf8')
```
Comment (thlorenz) [inline test/transform-stdout.js]: same as above

**Comment by thlorenz** [comment]
Granted final nits will be fixed LGTM now!
@jez are you OK with this as well? If so we can merge to master and publish a new minor.

**Comment by RichardLitt** [comment]
Fixed.

**Comment by thlorenz** [comment]
Cool, thanks @RichardLitt! Giving @jez a bit to give his verdict, but we'll most likely merge by EOD.

**Comment by jez** [comment]
LGTM. I've actually secretly wanted this feature for a while :D

**Comment by jez** [comment]
Also, :+1: for tests for `doctoc.js`, not just `transform.js`. We don't
have any of those yet; it's great to see them getting started.

**Comment by thlorenz** [comment]
Thanks, went out as v1.2.0!

**Comment by RichardLitt** [comment]
Woo! Thanks!

**Pull request closed without merging**